### DISCUSSION TITLE
Curator does not seem to close logs-syslog-so indices.

### DISCUSSION THREAD

**Discussion by Maurice-De** [Security-Onion-Solutions/securityonion#11894] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

8

### RAM

16

### Storage for /

300gb

### Storage for /nsm

1.7TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hi, 

I have tried to search for this, but a lot of old version topics came across, but no recent ones.

We have a distributed setup but after a couple of months we noticed that the logs where on 30days/365days with curator disabled. 

So I enabled curator from the grid config and changed several 365days to 90days, but after that threshold was reached there was no flattening of the used disk space, it was still increasing. I noticed in elestic index management that the most logging was in logs-syslog-so and so I went looking for that in the curator logging but found nothing. Only a lot of logs that stated that actions where skipped because there was nothing to close. There are actions for this index in /opt/so/conf/curator/action/ but I cannot find what should run this action and why it is not running.

Does anyone know if this is a known issue or if there is a standard quick config/fix to configure curator. Or does anyone know what I can try or troubleshoot for this?

Regards,
Maurice

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by weslambert** [comment]
Let's try this:

- Edit the following files to contain logs-syslog-so-close.yml instead of logs-syslog-close.yml
   Example: https://github.com/Security-Onion-Solutions/securityonion/blob/d3802c1668edbdd8c2b93740f15ac4293948fae2/salt/curator/tools/sbin/so-curator-cluster-close#L29
  - /opt/so/saltstack/default/salt/curator/tools/sbin/so-curator-close
  - /opt/so/saltstack/default/salt/curator/tools/sbin/so-curator-cluster-close
- sudo salt-call state.apply curator queue=True
- Observe the behavior to see if it aligns with expectations

**Reply by weslambert** [reply]
You might need to do one more thing.

- Modify /opt/so/saltstack/default/salt/curator/files/action/logs-syslog-so-close.yml to include the following line under the options section.

allow_ilm_indices: True

- Restart Curator or apply the changes and see if it makes a difference.

**Comment by Maurice-De** [comment]
Hi, 
with the alteration it seems the action runs, but it still throws the message "Skipping action "close" due to empty list: <class 'curator.exceptions.NoIndices'>"

**Reply by Maurice-De** [reply]
I think I have fixed it, in the value of:
https://github.com/Security-Onion-Solutions/securityonion/blob/d3802c1668edbdd8c2b93740f15ac4293948fae2/salt/curator/files/action/logs-syslog-so-close.yml

I have changed:
`value: '^(.ds-logs-syslog-so.*)$'`

to:
`value: '^(logs-syslog-so).*$'`

Then it closes the index.
But now I am thinking is this the right way to cycle the logging, in elastic under index lifecycle policies, there are also options for syslog?

What is best practice for rotating/cycling the logging? And what changes in Elastic can be done in Elastic or should everything be changed in Security Onion for the Elastic setup?

**Reply by Maurice-De** [reply]
It is also possible that I misconfigured syslog but I have followed:
[https://docs.securityonion.net/en/2.4/syslog.html](https://docs.securityonion.net/en/2.4/syslog.html)

![security onion](https://github.com/Security-Onion-Solutions/securityonion/assets/98149947/d8574965-7eae-4449-8732-a8eccea98887)

**Reply by weslambert** [reply]
logs-syslog-so should be a data stream with backing indices like .ds-logs-syslog-so. If I recall correctly, this was fixed in later versions of Security Onion. ILM manages certain aspects of the data streams and indices, but it can't close indices, so we use Curator to do it.

All changes should be made through the SOC administration page. I would advise you update to the latest version of Security Onion. We are working on a fix to address closing indices.

**Reply by Maurice-De** [reply]
Ok, I will try to update our security onion environment, but how can I check if the backing indices are configured for logs-syslog-so?
What behavior should I expect? Should the logs-syslog-so shrink and be offloaded to .ds-logs-syslog-so***?

**Reply by weslambert** [reply]
The logs-syslog-so index should stay there, but a data stream should be created with the same name. and its backing indices should be named as .ds-logs-syslog-so-YYYY.MM.DD-00000X. 

You can view the actual indices in Kibana, or use so-elasticsearch-indices-list.

**Reply by Maurice-De** [reply]
I have run the update with SOUP to 2.4.30, but unfortunately still no .ds-logs-syslog-so-YYY.MM.DD-00000X indices.

**Reply by Maurice-De** [reply]
I see this issue which is the issue you are referring to?
[https://github.com/Security-Onion-Solutions/securityonion/issues/11209](https://github.com/Security-Onion-Solutions/securityonion/issues/11209)

I see that the "data_stream: {}" line exists, but do I need to reset, rename or refresh something, or should this be fixed automatically?(hopefully without data loss I hope).

**Reply by weslambert** [reply]
A new data stream should be created with backing indices once syslog data is ingested. 

You can see the backing indices with so-elasticsearch-indices-list, or you can view the data stream or backing indices in Stack Management in Kibana.

To be clear, was your original issue more of a question about how to close indices to implement best practices or was the disk filled?

**Reply by Maurice-De** [reply]
Sorry if my initial question/issue was with a wrong assumption from my end, let me explain where I am at the moment:
First we have installed a distributed deployment of security onion.
Second we have made Syslog work by enabling it and configure the firewall rules with the grid config page as described in the docs.
Third we have configured a couple of devices to use the security onion as syslog destination
And fourth I came to realize that it is a lot of data and we only want to keep like 90 days of data so that we don't need lots of TB's to host the logging.

So I went looking how to change the retention and came across curator. Enabled it and came to realize it did not delete or even close it for syslog data. And eventually found out that the specific syslog action was not run. And now that it runs I noticed that the logs-syslog-so is in 1 big index, and after the update to 2.4.30 I see the option that is described in the issue I referred to, but the stack management (index management) still looks the same as in the previous screenshot. So I do not see an index (with format like ".ds-logs-syslog-so-YYYY.MM.DD-00000X") with the backing datastream log-syslog-so. But there is still syslog data received and put in the big index.

So it started indeed with a different question then where we are now, but I think the single big logs-syslog-so index is an issue in our deployment at the moment. The logs-syslog-so is now around 560GB and has syslog data for around 100 days.

**Reply by kvkamin** [reply]
under Administration/Configuration/curator/elasticsearch/index_settings/logs-syslog-so
what are the values for close and delete?  I had a similar problem and I found a crazy value there like 73000.  I don't know if that was default or something I did.. Also, i don't think they are actually deleted for 24 hours...

K

**Reply by Maurice-De** [reply]
@kvkamin that was the first thing I did when I started with the retention settings. Close is at 60 and delete is at 90. But that only applies to the ".ds-logs-syslog-so-YYYY.MM.DD-00000X" which I cannot find in my deployment.

**Reply by weslambert** [reply, marked as answer]
I think you would need to re-index the logs-syslog-so index to another named index, or delete it for the new data stream to be created. Otherwise, Elastic will use the template for the old index and Logstash will continue writing to it, since it matches the logs-syslog-so name.

This script may work for you, but please test it before using it in production.

https://github.com/weslambert/securityonion-elastic-misc/blob/2.x/so-elasticsearch-reindex

For an index of that size it may take a very long time.

**Reply by Maurice-De** [reply]
Do you mean I need to reindex it so it has a different name and can be deleted when my retention is reached (and a new indices/datastream is created for the new data), or do you mean the reindex will create the indices with the datastream?

**Reply by Maurice-De** [reply]
I have tried the reindexing, unfortunately it did not go as hoped. The data from that index has been lost (it was not that of a big deal and we still have a snapshot in case we really really need it). But in the good news segment: I now see a ".ds-logs-syslog-so-YYYY.MM.DD-00000X" with a "logs-syslog-so" datastream. So I hope Curator can now properly close and delete the syslog indices on time. I have marked the script as answer, but it was a combination with the update from 2.4.10 to 2.4.30; the reindexing script; the added action in Curator.

I will close the discussion as soon as the first indices are being closed if that is ok?
Thank you very much for the help @weslambert

**Comment by Maurice-De** [comment]
Thank you very much for the help, the first index is moved to the cold phase, so I think the deletion will also work.
